mautrix_signal/matrix.py

```diff
@@ -100,6 +100,56 @@ async def handle_join(self, room_id: RoomID, user_id: UserID, event_id: EventID)
 
         await portal.handle_matrix_join(user)
 
+    async def handle_kick_ban(
+        self,
+        action: str,
+        room_id: RoomID,
+        user_id: UserID,
+        sender: UserID,
+        reason: str,
+        event_id: EventID,
+    ) -> None:
+        self.log.debug(f"{user_id} was {action} from {room_id} by {sender} for {reason}")
+        portal = await po.Portal.get_by_mxid(room_id)
+        if not portal:
+            return
+
+        if user_id == self.az.bot_mxid:
+            if portal.is_direct:
+                await portal.cleanup_and_delete()
+            return
+
+        sender = await u.User.get_by_mxid(sender, create=False)
+        if not sender:
```

Comment:
This should check if the user is logged in (or alternatively it should support using the relaybot)

```suggestion
        if not sender or not await sender.is_logged_in():
```